crates/eigen-cli/src/lib.rs

```diff
@@ -20,14 +41,161 @@ pub enum EigenAddressCliError {
     RpcError(RpcError<TransportErrorKind>),
 }
 
+/// Possible errors raised while executing egnkey commands
+#[derive(Error, Debug)]
+pub enum EigenKeyCliError {
+    #[error("file error")]
+    FileError(std::io::Error),
+    #[error("encription error")]
+    KeystoreError(KeystoreError),
+    #[error("BLS error")]
+    BLSError(BlsError),
+    #[error("serialization error")]
+    SerializationError(SerializationError),
+}
+
+/// Executes an `egnkey` subcommand.
+///
+/// # Arguments
+///
+/// * `subcommand` - An egnkey subcommand which can be `generate`, `convert` or `derive-operator-id`.
+///
+/// # Errors
+///
+/// - If the subcommand execution fails (`EigenKeyCliError`).
+pub fn execute_egnkey_subcommand(subcommand: EigenKeyCommand) -> Result<(), EigenKeyCliError> {
+    match subcommand {
+        EigenKeyCommand::Generate {
+            key_type,
+            num_keys,
+            output_dir,
+        } => KeyGenerator::from(key_type).generate(num_keys, output_dir),
+
+        EigenKeyCommand::ConvertECDSA {
+            private_key,
+            output_file,
+            password,
+        } => store(private_key.into(), output_file, password)
+            .map_err(EigenKeyCliError::KeystoreError),
+
+        EigenKeyCommand::DeriveOperatorId { private_key } => {
+            let operator_id =
+                derive_operator_id(private_key).map_err(EigenKeyCliError::BLSError)?;
+            println!("{}", operator_id);
```

Comment:
print remove  pending

Reply:
This prints are for showing the command's results, not for debugging. The same is done in the Go implementation. Should we change it?

Reply:
oh, ok ,can u add  a cfg(not test ) flag then, so it doesn't print on tests

Reply:
all good